lgc/state/PalMetadata.cpp

```diff
@@ -811,3 +834,16 @@ void PalMetadata::updateSpiShaderColFormat(ArrayRef<ColorExportInfo> exps, bool
   }
   setRegister(mmSPI_SHADER_COL_FORMAT, spiShaderColFormat);
 }
+
+// =====================================================================================================================
+// Fills the xglCacheInfo section of the PAL metadata with the given data.
+//
+// @param finalizedCacheHash : This is the finalized hash that will be used to look for this pipeline in caches.
+// @param version : This is the version of llpc that generated the hash.
+void PalMetadata::setFinalized128BitCacheHash(llvm::ArrayRef<uint8_t> finalizedCacheHash,
+                                              llvm::ArrayRef<uint32_t> version) {
+  assert(finalizedCacheHash.size() == 16 && "The finalized cache hash is expected to be 128-bits.");
+  msgpack::MapDocNode &xglCacheInfo = m_pipelineNode[Util::Abi::PipelineMetadataKey::XglCacheInfo].getMap(true);
```

Comment:
Wonder if we should also include a version field to this node? Or is it enough to probe the node for what keys are in it on the XGL side?

Reply:
I don't think we need another version.  The XGL side can look for which entries are present and act accordingly.